File: semana_6/Ejercicios/3-Animales/animales.cpp
#include <iostream>

class Animal {

    public:
    std::string nombre;
    int edad;


    Animal(std::string _nombre, int _edad) {
        nombre = _nombre;
        edad = _edad;

    }

    void sonido() {
        std::cout << "El Animal llamado que se llama " << nombre << "hace un sonido"<< std::endl;
    }
};

class Perro : Animal {

    public:

    Perro(std::string _nombre, int _edad) : Animal(_nombre, _edad) {

    }

    void sonido() {
        std::cout << "El Animal llamado que se llama " << nombre << "hace un wau wau wau"<< std::endl;
    }
};

class Gato : Animal {

    public:
    Gato(std::string _nombre, int _edad) :  Animal(_nombre, _edad){

    }

    void sonido() {
        std::cout << "El Animal llamado que se llama " << nombre << "hace un sonido miau miau miau"<< std::endl;
    }
};

class Pajaro : Animal {

    public:

    Pajaro(std::string _nombre, int _edad) : Animal(_nombre,_edad){

    }

    void sonido() {
        std::cout << "El Animal llamado que se llama " << nombre << "hace un sonido piu piu"<< std::endl;
    }
};





int main(){

    Pajaro gorrion("Pablo",2);
    gorrion.sonido();

}

File: semana_6/Ejercicios/3-Animales/animales.py


class Animal():

    def __init__(self,nombre,edad):
        self.nombre = nombre
        self.edad = edad
    def Sonido(self):
        print(f"{self.nombre} hace un sonido")

class Perro(Animal):

    def __init__(self,nombre,edad):
        super().__init__(nombre,edad)
    
    def Sonido(self):
        print(f"{self.nombre} dice wauuu")

class Gato(Animal):

    def __init__(self,nombre,edad):
        super().__init__(nombre,edad)
    
    def Sonido(self):
        print(f"{self.nombre} dice miau")


class Pajaro(Animal):

    def __init__(self,nombre,edad):
        super().__init__(nombre,edad)
    
    def Sonido(self):
        print(f"{self.nombre} dice pio pio pio")


__